analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs

```diff
@@ -0,0 +1,23 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
```

Comment:
Let's add also
* System
* System.Collections.Generic
* System.Collections.Immutable
* System.Linq
* Microsoft.CodeAnalysis
* SonarAnalyzer.Extensions (there are not too many yet, but should be increasing while we move stuff from helpers)
* SonarAnalyzer.Helpers
* Moq

Reply:
And `Microsoft.CodeAnalysis.Diagnostics`

Reply:
These are just from top of my head. Did you try to write a small scraper that would measure what are the most often used ones? (Except C# and VB specific)

Reply:
Oh, and the `SonarAnalyzer.UnitTest.MetadataReferences`

Reply:
> These are just from top of my head. Did you try to write a small scraper that would measure what are the most often used ones? (Except C# and VB specific)

using System.Security; is used 42 times.
using System.Web; is used 46 times.
using System.Xml.XPath; is used 46 times.
using Microsoft.AspNetCore.Http; is used 46 times.
using System.ComponentModel; is used 48 times.
using Xunit; is used 56 times.
using Microsoft.AspNetCore.Mvc; is used 61 times.
using System.Net; is used 67 times.
using NUnit.Framework; is used 68 times.
using System.Collections.Immutable; is used 73 times.
using System.Globalization; is used 74 times.
using System.Collections.ObjectModel; is used 79 times.
using SonarAnalyzer.Common; is used 89 times.
using System.Runtime.CompilerServices; is used 90 times.
using SonarAnalyzer.Helpers; is used 95 times.
using System.Reflection; is used 96 times.
using Microsoft.VisualStudio.TestTools.UnitTesting; is used 106 times.
using SonarAnalyzer.UnitTest.MetadataReferences; is used 121 times.
using System.Threading; is used 123 times.
using System.Security.Cryptography; is used 131 times.
using CS = SonarAnalyzer.Rules.CSharp; is used 132 times.
using VB = SonarAnalyzer.Rules.VisualBasic; is used 133 times.
using System.Diagnostics; is used 139 times.
using System.Xml; is used 145 times.
using Microsoft.CodeAnalysis; is used 149 times.
using System.Runtime.Serialization; is used 154 times.
using System.Runtime.InteropServices; is used 173 times.
using System.Collections; is used 243 times.
using SonarAnalyzer.Rules.CSharp; is used 268 times.
using System.Text; is used 281 times.
using System.IO; is used 399 times.
using System.Threading.Tasks; is used 493 times.
using System.Linq; is used 727 times.
using System.Collections.Generic; is used 1970 times.
using System; is used 4495 times.

Reply:
This doesn't seem right, it probably includes TestCases directory too. `System.Runtime.Serialization` was found only 32x by VS for example.

Reply:
`System.Threading.Tasks` is 493x above and only 4x in the project. If you have some utility for it, you can do something like `code.Substring(0, code.IndexOf("namespace"))`

Reply:
> Let's add also
> 
> * System
> * System.Collections.Generic
> * System.Collections.Immutable
> * System.Linq
> * Microsoft.CodeAnalysis
> * SonarAnalyzer.Extensions (there are not too many yet, but should be increasing while we move stuff from helpers)
> * SonarAnalyzer.Helpers
> * Moq

Added them except SonarAnalyzer.Extensions and Moq, as those two would create ambiguity between some other classes and methods from other namespaces with the same names.